### DISCUSSION TITLE
Welcome to EmmyLuaCodeStyle Discussions!

### DISCUSSION THREAD

**Discussion by CppCXY** [CppCXY/EmmyLuaCodeStyle#126] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by edwinjhlee** [comment]
Hi everyone~ I'm working on developing x-cmd~ https://x-cmd.com

Recently I added a lua module to X-CMD. I was a user of node/npm. So referring to the habits of npm and go, users can use the lua module to realize:

`x lua install <pkg> # install pkg with luarocks`
`x lua fmt *.lua # use emmyluacodestyle to format lua code`
etc. 

I chose EmmyLuaCodeStyle as the default fmt because vscode development used EmmyLua for https://github.com/LuaLS/lua-language-server. I believe it is best for `x lua fmt` to be consistent with the scheme used by the vscode plugin.

I haven't had the time to spend using lua myself. Currently, we only provide the download of portable versions of lua and luarocks under the x-cmd pkg system. And we provide some possible simplifications via the lua module.


--------

大家好 ～ 我正在 开发 x-cmd ~ https://x-cmd.com

最近，我在 X-CMD 里面 加入了 lua 模块，我之前是一个 node/npm 的 用户。因此 参照 npm 和 go 的习惯，lua 模块可以让用户实现:

`x lua install <pkg> # using luarocks 安装 pkg`
`x lua fmt *.lua # using emmyluacodestyle 来 格式化 lua 代码`
等等

我之所以 选择 EmmyLuaCodeStyle 作为 默认的 fmt，是因为 vscode 开发中 https://github.com/LuaLS/lua-language-server 采用了 EmmyLua。而我认为最好还是 让 `x lua fmt` 与 vscode 的插件所采用的 方案 保持一致。

我本身还没来得及 花时间 去 使用 lua；所以只能算是 lua 的门外汉。当前我们仅在 x-cmd pkg 体系下 提供了 lua 和 luarocks 在 portable 版本下载。并通过 lua 模块提供一些 力所能及的 简化。
